terraform/modules/scheduler/main.tf

```diff
@@ -27,8 +27,29 @@ resource "google_cloud_scheduler_job" "poller_job" {
 
   pubsub_target {
     topic_name = var.pubsub_topic
-    data       = var.pubsub_data
+    data       = base64encode(jsonencode([
+      merge ({
+        "projectId" : "${var.project_id}",
+        "instanceId" : "${var.instance_id}",
+        "scalerPubSubTopic" : "${var.target_pubsub_topic}",
+        "units" : "PROCESSING_UNITS",
+        "minSize" : 100,
+        "maxSize" : 2000,
+        "scalingMethod" : "LINEAR",
+        "stateDatabase": var.terraform_spanner_state ? {
+          "name":       "spanner",
+          "instanceId": "${var.instance_id}",
```

Comment:
As a best practice shouldn't the state be stored in a different spanner instance than the one that is being autoscaled, so should this be separated out into a different variable.